### PR TITLE
feat: FindAllFiles, FindAllFilesExcept methods for ignoring .gitignore

### PR DESCRIPTION
**Pull request by maaslalani**
This PR introduces two new methods:

* `FindAllFiles`
* `FindAllFilesExcept`

to allow ignoring .gitignore when applicable.

### CODE REVIEW

**Review by caarlos0 — APPROVED** [review]

File: gitcha_test.go [hunk 1]
```diff
@@ -30,6 +30,40 @@ func TestGitRepoForPath(t *testing.T) {
...
+	tt := []struct {
+		path string
+		list []string
+		exp  string
+	}{
+		{".", []string{"*.test"}, "ignore.test"},
+	}
+
+	for _, test := range tt {
+		_, err := os.Create(test.exp)
```
Comment (muesli) [inline gitcha_test.go:43]: I think it might be nicer to do this in a temp dir: https://pkg.go.dev/io/ioutil#TempDir

Comment (caarlos0) [reply gitcha_test.go:43]: even better with `t.TempDir()` https://pkg.go.dev/testing@go1.19.1#T.TempDir

Comment (maaslalani) [reply gitcha_test.go]: Done!

File: gitcha_test.go [hunk 2]
```diff
@@ -30,6 +31,52 @@ func TestGitRepoForPath(t *testing.T) {
 	}
 }
 
+func TestFindAllFiles(t *testing.T) {
+	tmp := t.TempDir()
+
+	gitignore, err := os.Create(path.Join(tmp, ".gitignore"))
```
Comment (muesli) [inline gitcha_test.go:37]: I think `filepath.Join` is more appropriate here, e.g. for platforms like Windows.

Comment (maaslalani) [reply gitcha_test.go]: Fixed!

**Comment by muesli** [comment]
Thanks @maaslalani!

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -5,19 +5,19 @@ jobs:
   coverage:
     strategy:
       matrix:
-        go-version: [1.15.x]
+        go-version: [^1]
```
Comment (caarlos0) [inline .github/workflows/coverage.yml:8]: ps: can use `stable` to get the last stable go version

**Pull request merged**